File: BackEnd_SpringBoot/PockerPlanning/src/main/java/tn/esprit/pockerplanning/entities/commentaire.java
package tn.esprit.pockerplanning.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class commentaire {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    Long IdCommentaire;
    String contenue;
    @ManyToOne
    @JoinColumn(name="user_id")
    public User user;
    @ManyToOne
    @JoinColumn(name="id_activity")
    public Activity activity;

}

File: PockerPlanning_Ang/src/app/commentaire.ts
import { User } from "./core/models/user";
import { Activity } from "./modules/developpeur/views/list-activity/list-activity.component";

export class Commentaire {
    idcommentaire!:number;
    contenue!:string;
    user_id?: number; // ID de l'utilisateur qui a commenté
    id_activity?: number;
    user!: User;
    activity!:Activity;
}
